sema4ai/src/sema4ai_code/robocorp_language_server.py

```diff
@@ -1722,6 +1723,46 @@ def _get_runbook_path_from_agent_spec(
             result=runbook_path,
         ).as_dict()
 
+    @command_dispatcher(commands.SEMA4AI_REFRESH_AGENT_SPEC_INTERNAL)
+    def _refresh_agent_spec(self, params: RefreshAgentSpecDict) -> ActionResult:
+        action_spec_path = Path(params["action_spec_path"])
```

Comment:
I think that in this API we should receive the `agent-spec.yaml` and not the path to the action.

Point being that while we sometimes may call this after an action is created, the most common use-case for this will be calling this from a code-action.

Then, based on the `action-spec.yaml` you can get the parent and then list the actions below it.

As a note, I'm actually doing something that needed to list the actions below the agent root folder too, so, maybe you can just copy that for now. The code is below (probably needs some minor adjustments there for your use case -- probably should be a helper function we both use though, but maybe we can do private stuff ourselves in each branch and refactor in the future).

```

@dataclass
class _ActionPackageInFilesystem:
    relative_path: str
    package_yaml_path: Path

class XXX:

    def _list_existing_actions_in_filesystem(
        self,
    ) -> dict[Path, _ActionPackageInFilesystem]:
        found: dict[Path, _ActionPackageInFilesystem] = {}
        actions_dir: Path = (self._agent_root_dir / "actions").absolute()
        for package_yaml in actions_dir.rglob("package.yaml"):
            package_yaml = package_yaml.absolute()
            relative_path = str(package_yaml.relative_to(actions_dir).as_posix())
            found[package_yaml] = _ActionPackageInFilesystem(
                relative_path=relative_path, package_yaml_path=package_yaml
            )

        return found
```